OloEngine/src/OloEngine/Scene/SceneSerializer.cpp

```
@@ -123,6 +123,70 @@ namespace OloEngine
+        out << YAML::Key << "NoiseIntensity" << YAML::Value << fog.NoiseIntensity;
+        out << YAML::Key << "EnableLightShafts" << YAML::Value << fog.EnableLightShafts;
+        out << YAML::Key << "LightShaftIntensity" << YAML::Value << fog.LightShaftIntensity;
+        out << YAML::EndMap;
+    }
+
+    static void DeserializeFogSettings(const YAML::Node& data, FogSettings& fog)
+    {
+        if (auto fogNode = data["FogSettings"]; fogNode)
+        {
+            TrySet(fog.Enabled, fogNode["Enabled"]);
+            int mode = static_cast<int>(fog.Mode);
```

Clang (14.0.6)

[warning] variable 'mode' is not initialized

(cppcoreguidelines-init-variables)